ninja: fatal: invalid -j parameter

After cloning,
**git clone https://github.com/PX4/Firmware.git
**
and running,
no_sim=1 make px4_sitl gazebo

I am receiving the following error:
-- Found GStreamer: adding gst_camera_plugin
-- Found GStreamer: adding gst_video_stream_widget
-- Configuring done
-- Generating done
-- Build files have been written to: /home/boon/src/Firmware/build/px4_sitl_default/build_gazebo
[650/833] Performing build step for 'sitl_gazebo'
ninja: fatal: invalid -j parameter
[657/833] Building CXX object src/modu...odules__mavlink.dir/mavlink_main.cpp.o
FAILED: external/Stamp/sitl_gazebo/sitl_gazebo-build 
cd /home/boon/src/Firmware/build/px4_sitl_default/build_gazebo && /usr/bin/cmake --build /home/boon/src/Firmware/build/px4_sitl_default/build_gazebo -- -j 0
[663/833] Building CXX object src/modu...odules__mavlink.dir/mavlink_main.cpp.o
ninja: build stopped: subcommand failed.
Makefile:230: recipe for target 'px4_sitl' failed
make: *** [px4_sitl] Error 1

> After cloning, **git clone https://github.com/PX4/Firmware.git ** and running, no_sim=1 make px4_sitl gazebo
> 
> I am receiving the following error: -- Found GStreamer: adding gst_camera_plugin -- Found GStreamer: adding gst_video_stream_widget -- Configuring done -- Generating done -- Build files have been written to: /home/boon/src/Firmware/build/px4_sitl_default/build_gazebo [650/833] Performing build step for 'sitl_gazebo' ninja: fatal: invalid -j parameter [657/833] Building CXX object src/modu...odules__mavlink.dir/mavlink_main.cpp.o FAILED: external/Stamp/sitl_gazebo/sitl_gazebo-build cd /home/boon/src/Firmware/build/px4_sitl_default/build_gazebo && /usr/bin/cmake --build /home/boon/src/Firmware/build/px4_sitl_default/build_gazebo -- -j 0 [663/833] Building CXX object src/modu...odules__mavlink.dir/mavlink_main.cpp.o ninja: build stopped: subcommand failed. Makefile:230: recipe for target 'px4_sitl' failed make: *** [px4_sitl] Error 1

same problem no solution on all web

Hi there. I am having the same issue.
Env: Ubuntu 18.04 on VMware

> Oh, it looks like the argument `0` is not allowed for Ninja.
> 
> Could you change `DEBUG` to `FATAL_ERROR` here:
> 
> https://github.com/PX4/PX4-Autopilot/blob/2870b5bbb167d0571d22790d4c350600fbffe684/platforms/posix/cmake/sitl_target.cmake#L52-L53
> 
> and then paste the output here?

After I change `DEBUG` to `FATAL_ERROR`, I got the output:

-- PX4 version: v1.13.0-alpha1-4065-g0ce44cebb7
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.6.9", minimum required is "3") 
-- PX4 config file: /home/james/PX4-Autopilot/boards/px4/sitl/default.px4board
-- PLATFORM posix
-- ROMFSROOT px4fmu_common
-- TESTING y
-- ETHERNET y
-- PX4 config: px4_sitl_default
-- PX4 platform: posix
-- PX4 lockstep: enabled
-- cmake build type: RelWithDebInfo
-- The CXX compiler identification is GNU 7.5.0
-- The C compiler identification is GNU 7.5.0
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/cc
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Building for code coverage
-- ROMFS: ROMFS/px4fmu_common
CMake Error at platforms/posix/cmake/sitl_target.cmake:52 (message):
  2 logical cores detected and 1372 megabytes of memory available.

  Limiting sitl_gazebo and simulation-ignition concurrent jobs to 0
Call Stack (most recent call first):
  platforms/posix/CMakeLists.txt:121 (include)


-- Configuring incomplete, errors occurred!
See also "/home/james/PX4-Autopilot/build/px4_sitl_default/CMakeFiles/CMakeOutput.log".
Error: /home/james/PX4-Autopilot/build/px4_sitl_default is not a directory
Makefile:235: recipe for target 'px4_sitl' failed
make: *** [px4_sitl] Error 1

--- a/platforms/posix/cmake/sitl_target.cmake
+++ b/platforms/posix/cmake/sitl_target.cmake
@@ -49,6 +49,10 @@ if(parallel_jobs GREATER NUMBER_OF_LOGICAL_CORES)
 	set(parallel_jobs ${NUMBER_OF_LOGICAL_CORES})
 endif()
 
+if(parallel_jobs LESS 1)
+	set(parallel_jobs 1)
+endif()
+
 message(DEBUG  "${NUMBER_OF_LOGICAL_CORES} logical cores detected and ${AVAILABLE_PHYSICAL_MEMORY} megabytes of memory available.
 		Limiting sitl_gazebo and simulation-ignition concurrent jobs to ${parallel_jobs}")
 

